tools/ClassGen/NodeGen.cpp

```diff
@@ -386,13 +386,12 @@ int main(int argc, char **argv) {
   BB.newNode("SGD")
       .addInput("Gradient")
       .addInput("Weight")
-      .addInput("Gsum")
       .addMember(MemberType::Float, "L1Decay")
       .addMember(MemberType::Float, "L2Decay")
       .addMember(MemberType::Float, "LearningRate")
       .addMember(MemberType::Float, "Momentum")
       .addMember(MemberType::Unsigned, "BatchSize")
-      .addOverwrittenInput("Weight")
+      .addResult("Weight.getType()", "UpdatedWeight")
```

Comment:
Is the result the updated weight or the delta that we need to add to the weight? We need to clarify that in a comment, I think.